### PR TITLE
Maeste/issue137

### PR DESCRIPTION
**Pull request by maeste**
#137 Basic configurationi for mkdocs, Automatic reference generation from dcustring
Add general documentation to the project #142

### CODE REVIEW

File: docs/multi_agents.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# What are AI Agents?
+AI agents are autonomous software systems that can perceive their environment, make decisions, and take actions to achieve specific goals. They are designed to operate without human intervention, leveraging advanced AI techniques like natural language processing and machine learning.
+## Key characteristics of AI agents
+* **Autonomy**: AI agents can make decisions and act independently based on their goals and the information they gather.
+* **Flexibility**: They can adapt to changing circumstances and learn from experience to improve their performance over time.
+* **Reactivity**: AI agents can perceive their environment and respond to changes in real-time.
+
+## Components of an AI Agent
+AI agents are composed of several key components that enable their autonomous behavior:
+Sensors: Used to gather information from the environment, such as user inputs, sensor data, or external APIs.
```
Comment (kabir) [inline docs/multi_agents.md:10]: I think perhaps use an unordered list here. Possibly with the keywords in bold

File: docs/multi_agents.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+* **Communication Mechanism**:
+    * Direct Communication: Agents send messages to each other.
+    * Indirect Communication: Agents use shared context or memory.
+* **Coordination and Cooperation**:
+Strategies for agents to work together towards common goals, avoid conflicts, and optimize joint performance.
+    * Examples:
+        * Task Allocation: Dividing tasks among agents based on their capabilities.
+        * Task planning and prioritization
+        * Negotiation/reasoning: Agents discuss to reach agreements.
+        * Conflict resolution: Avoid loop or infinite loop for the same question
```
Comment (kabir) [inline docs/multi_agents.md:32]: s/loop/loops

File: docs/multi_agents.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        * Challenge other agents: challenge previous steps to reduce hallucinations
+        * Decision-Making: Methods for agents to make choices based on their goals, perceptions, and available information. LLMs or other AI algorithms (tule engine, game theory etc) can be used.
+* **Retrieve additional information**: RAG agents can retrieve informations for better results. The agentic approach permit also a multi-step RAG refining progressively the information retrieved
```
Comment (kabir) [inline docs/multi_agents.md:35]: s/informations/information

And, "The agentic approach also permits a multi-step..." sounds better :-)

File: GETTING_STARTED.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
+# wise-agents getting started
+
+# How to resolve dependencies declared in pyptoject.toml
+
+1. Create a venv and activate it
+``` 
+python -m venv .venv
+source .venv/bin/activate
+```
+2. ``pip install .``
```
[the remark this answers is not in the document]
Comment (maeste) [reply GETTING_STARTED.md:10]: You are right. My plan is to change this getting started completely (maybe giving a step by step guide to run a simple example) as soon as the CONTRIBUTING.md is ok

File: GETTING_STARTED.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
+# wise-agents getting started
+
+# How to resolve dependencies declared in pyptoject.toml
```
Comment (fjuma) [inline GETTING_STARTED.md:3]: s/pyptoject.toml/pyproject.toml

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+* Python 3.12+
+* Git
+* An [IDE](https://en.wikipedia.org/wiki/Comparison_of_integrated_development_environments#Python)
+(e.g., [Microsoft Visual Studio Code](https://code.visualstudio.com/))
+
+To setup your development environment you need to:
+
+1. First `cd` to the directory where you cloned the project (eg: `cd wise-agents`)
+
+2. Create a Python wirtual environment for the project.
```
Comment (fjuma) [inline CONTRIBUTING.md:59]: s/wirtual/virtual

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+    ```
+
+
+## Contributing Guidelines
+
+When submitting a PR, please keep the following guidelines in mind:
+
+1. In general, it's good practice to squash all of your commits into a single commit. For larger changes, it's ok to have multiple meaningful commits. If you need help with squashing your commits, feel free to ask us how to do this on your pull request. We're more than happy to help!
+
+2. Please include the issue you worked on in the title of your pull request and in your commit message. For example, for [issue-125](https://github.com/wise-agents/wise-agents/issues/125), the PR title and commit message should be `Go through TODOs in the code and create issues for them #125`.
```
Comment (fjuma) [inline CONTRIBUTING.md:102]: For GitHub issues, unlike the standard practice with JIRAs, it seems that most projects don't ask for the issue number in the commit message/title.

Instead, a pull request can be linked to an issue using a keyword (as described here: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

So maybe we can reword 2. to something like:

Please link the issue you worked on in the description of your pull request and in your commit message. For example, for [issue-125](https://github.com/wise-agents/wise-agents/issues/125), the PR description and commit message could be: 

```
Go through TODOs in the code and create issues for them

Fixes #125
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+
+## Code Reviews
+
+All submissions, including submissions by project members, need to be reviewed by at least one wise-agents committer before being merged.
+
+The [GitHub Pull Request Review Process](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/about-pull-request-reviews) is followed for every pull request.
+
+
+## Community
+For more information on how to get involved with WildFly Elytron, check out our [community](https://wildfly-security.github.io/wildfly-elytron/community/) page.
```
Comment (fjuma) [inline CONTRIBUTING.md:113]: s/WildFly Elytron/Wise Agents :)

Comment (fjuma) [reply CONTRIBUTING.md:113]: The URL would need to be removed/updated too.

File: README.md [hunk 9]
```diff
@@ -1,30 +1,16 @@
-# wise-agents
+# wise-agents a distributed multi AI agent framework
```
Comment (fjuma) [inline README.md:1]: For readability, maybe "Wise Agents: A Distributed AI Multi-Agent Framework" would be better.

File: README.md [hunk 10]
```diff
@@ -1,30 +1,16 @@
...
 
-# How to resolve dependencies declared in pyptoject.toml
+Wise-agent is a framework implementing an architecture of distribute multi AI agents.
```
Comment (fjuma) [inline README.md:3]: Wise-agents

s/distribute/distrubted

**Review by fjuma — APPROVED** [review]

**Pull request merged**